incubator/virtualcluster/pkg/syncer/conversion/mutate.go

```diff
@@ -252,12 +249,12 @@ func getServiceEnvVarMap(ns, cluster string, enableServiceLinks *bool, services
 		}
 		serviceName := service.Name
 
-		// We always want to add environment variabled for master services
+		// We always want to add environment variables for master services
 		// from the corresponding master service namespace of the virtualcluster,
 		// even if enableServiceLinks is false.
 		// We also add environment variables for other services in the same
 		// namespace, if enableServiceLinks is true.
-		if service.Namespace == tenantMasterSvcNs && masterServices.Has(serviceName) {
+		if IsControlPlaneService(service, cluster) {
```

Comment:
It seems that the services will not contain the services in the $cluster namespace. You may need to change "getPodRelatedServices" method in pod/dws.go as well.

Reply:
Oops, you are right, I was doing that in a follow up PR and forgot it needed it for this one…

Reply:
Resolved in latest commit.